compiler/rustc_middle/src/hir/mod.rs

```diff
@@ -149,6 +149,7 @@ pub fn provide(providers: &mut Providers) {
         index.parenting.get(&id).copied().unwrap_or(CRATE_HIR_ID)
     };
     providers.hir_attrs = |tcx, id| AttributeMap { map: &tcx.untracked_crate.attrs, prefix: id };
+    providers.source_span = |tcx, def_id| tcx.definitions.def_span(def_id).with_parent(None);
```

Comment:
It looks like the `with_parent(None)` part would make it impossible for the sanity check in the `track_span_parent` callback to ever fail. Is this intended? Or should we rather check here if `tcx.definitions.def_span` returns a span without parent?